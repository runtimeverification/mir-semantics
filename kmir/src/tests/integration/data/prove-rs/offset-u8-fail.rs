// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT

// https://github.com/model-checking/kani/blob/main/tests/expected/offset-u8-fail/main.rs

// Checks that the pointer computed with `offset` causes a failure if it's
// dereferenced outside the object bounds
#![feature(core_intrinsics)]
use std::intrinsics::offset;

fn main() {
    test_offset();
}

// #[kani::proof]
fn test_offset() {
    let s: &str = "123";
    let ptr: *const u8 = s.as_ptr();

    unsafe {
        let x = *offset(ptr, 3isize) as char;
    }
}
