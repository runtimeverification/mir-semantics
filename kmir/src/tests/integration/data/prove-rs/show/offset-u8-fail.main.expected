
┌─ 1 (root, init)
│   #execTerminator ( terminator ( ... kind: terminatorKindCall ( ... func: operandC
│   span: 0
│
│  (74 steps)
└─ 3 (stuck, leaf)
    #traverseProjection ( toLocal ( 5 ) , thunk ( #applyBinOp ( binOpOffset , thunk
    span: 70


┌─ 2 (root, leaf, target, terminal)
│   #EndProgram ~> .K


